### PR TITLE
[Merged by Bors] - feat: log connector name and version on startup

### PR DESCRIPTION
**Pull request by EstebanBorai**
Provides support to log connector name and version on startup.

---

## Demo

```bash
2023-06-28T17:34:59.069598Z  INFO json_test_connector: Starting Processing my-json-test-connector@0.1.0
```

> `<pkg_name>@<version>`

```bash
➜  json-test-connector git:(3334-log-connector-version-string-on-connector-startup) ✗ cdk deploy log --name my-json-test-connector
2023-06-28T17:34:59.068577Z  INFO json_test_connector: Using EnvSecretStore
2023-06-28T17:34:59.068703Z  INFO json_test_connector: Reading config file from: /json-test-connector/sample-config.yaml
2023-06-28T17:34:59.069598Z  INFO json_test_connector: Starting Processing my-json-test-connector@0.1.0
2023-06-28T17:34:59.071519Z  INFO fluvio::config::tls: Using verified TLS with inline certificates domain="bitter-dawn-1012b4378e492dd6c3b4f99911f5a371.c.dev.infinyon.cloud"
2023-06-28T17:34:59.075361Z  INFO fluvio::fluvio: Connecting to Fluvio cluster fluvio_crate_version="0.19.2" fluvio_git_hash="7432c94b606b0a3a820a2f499d4d397728e96c47"
2023-06-28T17:35:00.193867Z  INFO connect: fluvio_socket::versioned: connect to socket add=router.dev.infinyon.cloud:9003
2023-06-28T17:35:00.759571Z  INFO connect:connect_with_config: fluvio::config::tls: Using verified TLS with inline certificates domain="bitter-dawn-1012b4378e492dd6c3b4f99911f5a371.c.dev.infinyon.cloud"
2023-06-28T17:35:01.469555Z  INFO connect:connect_with_config:connect: fluvio_socket::versioned: connect to socket add=router.dev.infinyon.cloud:9003
2023-06-28T17:35:03.255467Z  INFO dispatcher_loop{self=MultiplexDisp(8)}: fluvio_socket::multiplexing: multiplexer terminated
2023-06-28T17:35:04.095361Z  INFO fluvio_connector_common::monitoring: using default metric path: /tmp/fluvio-connector.sock
2023-06-28T17:35:04.095688Z  INFO fluvio_connector_common::monitoring: metric file already exists, deleting: /tmp/fluvio-connector.sock
producing a value: {"template":"test"}
2023-06-28T17:35:04.097975Z  INFO fluvio_connector_common::monitoring: monitoring started
2023-06-28T17:35:05.096034Z  INFO run:create_serial_socket_from_leader{leader_id=0}:connect_to_leader{leader=0}:connect: fluvio_socket::versioned: connect to socket add=router.dev.infinyon.cloud:9005
producing a value: {"template":"test"}
producing a value: {"template":"test"}
2023-06-28T17:35:19.098178Z  INFO run: fluvio::producer::partition_producer: partition producer end event received
```

### CODE REVIEW

File: crates/fluvio-connector-derive/src/generator.rs [hunk 1]
```diff
@@ -125,7 +125,7 @@ fn init_and_parse_config(config_type_path: &Path) -> TokenStream {
 
         let user_config: #config_type_path = ::fluvio_connector_common::config::from_value(config_value, Some(#config_type_path::__config_name()))?;
 
-        ::fluvio_connector_common::tracing::info!("starting processing");
+        ::fluvio_connector_common::tracing::info!("Starting Processing {}@{}", common_config.name(), common_config.version());
```
Comment (morenol) [inline crates/fluvio-connector-derive/src/generator.rs:128]: I think that we should tracing format:

```
  info!(connector_version = % common_config.version(), "Starting processing")
```

**Review by morenol — CHANGES_REQUESTED** [review]
I think that more important than the name would be to log the connector type. We should add it to the log

**Comment by EstebanBorai** [comment]
> I think that more important than the name would be to log the connector type. We should add it to the log

Just updated approach!

```
2023-06-28T19:56:46.543228Z  INFO json_test_connector: Starting Processing conn_type="json-test-source" conn_version="my-json-test-connector" conn_version="0.1.0"
2023-06-28T19:56:46.544752Z  INFO fluvio::config::tls: Using verified TLS with inline certificates domain="bitter-dawn-1012b4378e492dd6c3b4f99911f5a371.c.dev.infinyon.cloud"
2023-06-28T19:56:46.549094Z  INFO fluvio::fluvio: Connecting to Fluvio cluster fluvio_crate_version="0.19.2" fluvio_git_hash="7432c94b606b0a3a820a2f499d4d397728e96c47"
2023-06-28T19:56:47.699133Z  INFO connect: fluvio_socket::versioned: connect to socket add=router.dev.infinyon.cloud:9003
2023-06-28T19:56:48.606864Z  INFO connect:connect_with_config: fluvio::config::tls: Using verified TLS with inline certificates domain="bitter-dawn-1012b4378e492dd6c3b4f99911f5a371.c.dev.infinyon.cloud"
2023-06-28T19:56:49.402115Z  INFO connect:connect_with_config:connect: fluvio_socket::versioned: connect to socket add=router.dev.infinyon.cloud:9003
2023-06-28T19:56:51.086602Z  INFO dispatcher_loop{self=MultiplexDisp(8)}: fluvio_socket::multiplexing: multiplexer terminated
2023-06-28T19:56:51.880764Z  INFO fluvio_connector_common::monitoring: using default metric path: /tmp/fluvio-connector.sock
producing a value: {"template":"test"}
2023-06-28T19:56:51.881228Z  INFO fluvio_connector_common::monitoring: metric file already exists, deleting: /tmp/fluvio-connector.sock
2023-06-28T19:56:51.883628Z  INFO fluvio_connector_common::monitoring: monitoring started
2023-06-28T19:56:52.904789Z  INFO run:create_serial_socket_from_leader{leader_id=0}:connect_to_leader{leader=0}:connect: fluvio_socket::versioned: connect to socket add=router.dev.infinyon.cloud:9005
producing a value: {"template":"test"}
```

**Comment by EstebanBorai** [comment]
@morenol this would be the new log format:

```bash
2023-06-28T19:56:46.543228Z  INFO json_test_connector: Starting Processing conn_type="json-test-source" conn_version="my-json-test-connector" conn_version="0.1.0"
```

File: crates/fluvio-connector-derive/src/generator.rs [hunk 2]
```diff
@@ -125,7 +125,7 @@ fn init_and_parse_config(config_type_path: &Path) -> TokenStream {
 
         let user_config: #config_type_path = ::fluvio_connector_common::config::from_value(config_value, Some(#config_type_path::__config_name()))?;
 
-        ::fluvio_connector_common::tracing::info!("starting processing");
+        ::fluvio_connector_common::tracing::info!(conn_type=common_config.r#type(), conn_version=common_config.name(), conn_version=common_config.version(), "Starting Processing");
```
Comment (morenol) [inline crates/fluvio-connector-derive/src/generator.rs:128]:
```suggestion
        ::fluvio_connector_common::tracing::info!(conn_type=common_config.r#type(), conn_name=common_config.name(), conn_version=common_config.version(), "Starting Processing");
```

**Review by morenol — APPROVED** [review]

**Pull request closed without merging**